src/net/gateway-archive.ts

```diff
@@ -0,0 +1,90 @@
+import { FileApi } from "../api/file";
+import { ContentUri } from "../wrappers/content-uri"
+import {
+    VOCDONI_MAINNET_ENTITY_ID,VOCDONI_RINKEBY_ENTITY_ID, VOCDONI_GOERLI_ENTITY_ID, VOCDONI_XDAI_ENTITY_ID, VOCDONI_SOKOL_ENTITY_ID, XDAI_ENS_REGISTRY_ADDRESS, XDAI_PROVIDER_URI, XDAI_CHAIN_ID,
+    SOKOL_CHAIN_ID, SOKOL_PROVIDER_URI, SOKOL_ENS_REGISTRY_ADDRESS, XDAI_STG_ENS_REGISTRY_ADDRESS, VOCDONI_XDAI_STG_ENTITY_ID
+} from "../constants"
+import { TextRecordKeys } from "../models/entity"
+import { IGateway } from "./gateway";
+import { DVoteGatewayResponseBody } from "./gateway-dvote";
+import { IGatewayPool } from "./gateway-pool";
+import { keccak256 } from "@ethersproject/keccak256"
+
+export type EthNetworkID = "mainnet" | "rinkeby" | "goerli" | "xdai" | "sokol"
+
+export class GatewayArchive {
+
+    /**
+     * Fetch the process data on the IPFS archive
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    public static getProcessFromArchive(processId: string, gateway: IGateway | IGatewayPool, errorMessage: string): Promise<DVoteGatewayResponseBody> {
```

Comment:
Why `errorMessage`? Why not a custom error class like we had started doing?

Reply:
How about `GatewayArchive.getProcess`?  The "FromArchive" feels redundant

Reply:
Also, we will need to use the new gateway client interfaces

Reply:
> Why `errorMessage`? Why not a custom error class like we had started doing?

In this case we want to give exactly the same error that would have been given by the caller function in case the process id wasn't found.

> How about `GatewayArchive.getProcess`? The "FromArchive" feels redundant

Sure

> Also, we will need to use the new gateway client interfaces

Yes, after the rebase